python/pyarrow/tests/test_fs.py

```diff
@@ -703,6 +703,10 @@ def test_get_file_info_with_selector(fs, pathfn):
             # on the s3fs/fsspec version combo, it includes the base_dir
             # (https://github.com/dask/s3fs/issues/393)
             assert (len(infos) == 4) or (len(infos) == 5)
+        elif fs.type_name in ["py::fsspec+file", 'py::fsspec+memory']:
+            # fsspec also lists root dir
+            # GH-37555
```

Comment:
Hmm, if fsspec does that, then we should probably work around it in our FSSpecHandler.

Reply:
If I understand correctly, you think we should work around it in a way that we do not include the root dir in the result?

Reply:
Yes. fsspec might change their semantics from time to time, but we should try to follow our own spec :-)

Reply:
This is a very important info, thanks! =)
Instead of updating the test I will work around it in FSSpecHandler 👍